### PR TITLE
add calc of cpi linked mortgage to amortization schedule

### PR DESCRIPTION
**Pull request by shamit6**

### CODE REVIEW

File: components/Mortgage/MortgageProgram.tsx [hunk 1]
```diff
@@ -202,7 +202,7 @@ export default function MortgageProgram({
                 onChange={(e) => {
                   setMortgageProgram({
                     ...mortgageProgram,
-                    earlyPayoffPurpose: e.target.value,
+                    earlyPayoffPurpose: e.target.value as MortgageEarlyPayoffPurpose,
```
Comment (deanshub) [inline components/Mortgage/MortgageProgram.tsx:205]: Is that int?

File: utils/types.ts [hunk 2]
```diff
@@ -50,19 +50,20 @@ export enum MortgageEarlyPayoffPurpose {
 
 export interface MortgageProgramData {
   amount: number
-  type?: MortgageType
-  returnType: string
+  type: MortgageType
+  returnType?: string
```
Comment (deanshub) [inline utils/types.ts]: Why optional?

Comment (shamit6) [reply utils/types.ts]: fixed

File: components/Mortgage/MortgageProgram.tsx (hunk 1 above)
Comment (shamit6) [reply components/Mortgage/MortgageProgram.tsx:205]: it's an enum:
```
enum MortgageEarlyPayoffPurpose {
  SHORTENING_DURATION = 'shortening-duration',
  REDUCINNG_PAYMENT = 'reducinng-payment',
}
```

Comment (shamit6) [reply components/Mortgage/MortgageProgram.tsx:205]: It's a select input

**Review by deanshub — APPROVED** [review]

**Pull request merged**